ethpm_cli/validation.py

```diff
@@ -0,0 +1,51 @@
+from argparse import Namespace
+from pathlib import Path
+
+from ethpm.typing import URI
+from ethpm.utils.ipfs import is_ipfs_uri
+from ethpm.utils.uri import is_valid_content_addressed_github_uri
+from ethpm.validation import is_valid_registry_uri, validate_package_name
+
+from ethpm_cli.exceptions import InstallError, UriNotSupportedError, ValidationError
+
+
+def validate_parent_directory(parent_dir: Path, child_dir: Path) -> None:
+    if parent_dir not in child_dir.parents:
+        raise InstallError(f"{parent_dir} was not found in {child_dir} directory tree.")
+
+
+def validate_cli_args(args: Namespace) -> None:
+    validate_target_uri(args.uri)
+    validate_alias(args.alias)
+    validate_ethpm_dir(args.ethpm_dir)
+
+
+def validate_target_uri(uri: URI) -> None:
+    if (
+        not is_ipfs_uri(uri)
+        and not is_valid_registry_uri(uri)  # noqa: W503
+        and not is_valid_content_addressed_github_uri(uri)  # noqa: W503
+    ):
+        raise UriNotSupportedError(
+            f"Target uri: {uri} not a currently supported uri. "
+            "Target uris must be one of: ipfs, github blob, or registry."
+        )
+
+
+def validate_alias(alias: str) -> None:
+    if alias:
```

Comment:
IIUC this `if` statement is to allow for a `None` value?

Two things.

1. This would allow the empty string.  Is that intentional?  Also other falsy values like `0` or `False`.  Probably not ideal.
2. I would advocate for this validation function to purely evaluate actual aliases and to leave the acceptance of `None` or other allowed empty values to happen at a higher level.  Same with `validate_ethpm_dir`